### ISSUE TITLE
Understanding model-view design

### ISSUE THREAD

**Issue by jekadwell** [PacktPublishing/Mastering-GUI-Programming-with-Python#5]
I could not get the code for the text-file editor to work.  This is the standard code for the text-file editor, not the model-view code.  I was getting a 'undefined variable' error for 'path' in the following:
path.exists(filename) .  The program would run, but would not recognize that the file exists nor would it report an error.  The program would just stop and not write the file or report and error other than something like this, 'Process finished with exit code -1073740791 (0xC0000409)'

I added from 'os import path' to the header to solve the problem.

**Comment by alandmoore** [comment]
Assuming we're talking about the file Chapter05/non-model-view_code.py, `from os import path` is line 2.

**Comment by jekadwell** [comment]
import sys

	
from PyQt5 import QtWidgets as qtw

	
from PyQt5 import QtGui as qtg

	
from PyQt5 import QtCore as qtc


 

 

 

 

Thanks, you are correct.  I mistakenly thought I was to use this from Chapter 4 and following the example in the book I did not see any reference to the import path code.  I did not look at the non-model view code.  I should have opened that link.  I had just started the section titled “Understanding model-view design”

 

Thanks, again.

John
